### PR TITLE
[CLIPSeg] Make interpolate_pos_encoding default to True

### LINKED ISSUE #34415: ClipSeg broken
### System Info

irrelevant, happens on any env with updated transformers, replicated myself on `Transformers v4.46.0` (current release on pip at time of writing)

### Who can help?

@amyeroberts @manuelsh 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [x] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

- `pip install transformers==4.46.0` (latest currently)
- Follow official documentation and examples for ClipSeg https://huggingface.co/docs/transformers/model_doc/clipseg#transformers.CLIPSegModel.forward.example

Here's a fully contained python file if you want a lazy replication:
```py
import torch
import numpy as np
from transformers import CLIPSegProcessor, CLIPSegForImageSegmentation

# Create gray image at 512x512 with 32x32 black spot in the middle for clipseg to find
# Note: intentionally use a size that isn't the ClipSeg size just to prove that the input size is irrelevant
arr = np.ones((512, 512, 3), dtype=np.uint8) * 128
arr[240:272, 240:272] = 0

processor = CLIPSegProcessor.from_pretrained("CIDAS/clipseg-rd64-refined")
model = CLIPSegForImageSegmentation.from_pretrained("CIDAS/clipseg-rd64-refined")

with torch.no_grad():
    processed = processor(text="the black spot", images=arr, return_tensors="pt", padding=True)
    print(f"{processed.pixel_values.shape}") # prints 1, 3, 352, 352
    mask = model(**processed)[0] # ValueError: Input image size (352*352) doesn't match model (224*224).

# Now do something with the mask, maybe save a PIL image,
# doesn't matter cause you can't get to here without an error

```

### Expected behavior

Expected behavior: it works, as it did before updating transformers

But instead, it fails with `Input image size (352*352) doesn't match model (224*224).`
See my comment on the PR that broke it: https://github.com/huggingface/transformers/pull/32600/files#r1816732699

Secondarily, there's now a warning about one of the arguments (That, again, is in the docs and has been there for years):
`UserWarning: The following named arguments are not valid for 'ViTImageProcessor.preprocess' and were ignored: 'padding'`

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

This PR fixes #34415 by making `interpolate_pos_encoding` default to `True`.

### CODE REVIEW

**Comment by manuelsh** [comment]
Please see my comment https://github.com/huggingface/transformers/issues/34415#issuecomment-2438972495 before merging.

**Comment by NielsRogge** [comment]
I've updated my PR description based on your comments

File: src/transformers/models/clipseg/modeling_clipseg.py [hunk 1]
```diff
@@ -164,62 +163,40 @@ def __init__(self, config: CLIPSegVisionConfig):
...
+            .T
+        )
+        result = torch.cat([self.position_embedding.weight[:1], b])
 
-        patch_pos_embed = patch_pos_embed.permute(0, 2, 3, 1).view(1, -1, dim)
-
-        return torch.cat((class_pos_embed, patch_pos_embed), dim=1)
+        return result
 
-    def forward(self, pixel_values: torch.FloatTensor, interpolate_pos_encoding=False) -> torch.Tensor:
```
Comment (ArthurZucker) [inline src/transformers/models/clipseg/modeling_clipseg.py:208]: Why not just revert this? I think we should keep what we had: script support and etc!

Comment (NielsRogge) [reply src/transformers/models/clipseg/modeling_clipseg.py:208]: I would prefer to keep the old way of interpolation to match the logits

Comment (ArthurZucker) [reply src/transformers/models/clipseg/modeling_clipseg.py:208]: THis is another kind of breaking change were we added support for torch script, and now we are removing it + we have less standardization. 

Can you investigate in the logit difference?

**Review by ArthurZucker — APPROVED** [review]
LGTM thanks, not sure why one of the integration test is gone, congrats on getting the old logits back! 🥳

**Comment by NielsRogge** [comment]
I just reverted to the interpolation which is used by all other models, so the logits diff remains

File: src/transformers/models/clipseg/modeling_clipseg.py (hunk 1 above)
Comment (NielsRogge) [reply src/transformers/models/clipseg/modeling_clipseg.py:208]: Yes it was just minor difference in my local setup!

**Pull request merged**